File: backend/models/Booking.js
const mongoose = require("mongoose");

const BookingSchema = new mongoose.Schema(
    {
        bookingId: { type: String, required: true, unique: true },

        userId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
        parkingId: { type: mongoose.Schema.Types.ObjectId, ref: "ParkingListing", required: true },

        vehicleType: {
            type: String,
            enum: ["TWO_WHEELER", "FOUR_SEATER", "SIX_SEATER", "SUV"],
            required: true,
        },

        startTime: { type: Date, required: true },
        endTime: { type: Date, required: true },

        totalHours: { type: Number, required: true },
        totalAmount: { type: Number, required: true },

        status: {
            type: String,
            enum: ["BOOKED", "CHECKED_IN", "CHECKED_OUT", "CANCELLED", "EXPIRED"],
            default: "BOOKED",
        },

        qrCodeDataUrl: { type: String },
    },
    { timestamps: true }
);

module.exports = mongoose.model("Booking", BookingSchema);

File: backend/src/main/java/com/parkease/model/Booking.java
package com.parkease.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document(collection = "bookings")
public class Booking {
    @Id
    private String id;

    @Indexed(unique = true)
    private String bookingId; // Human readable or short ID

    @DBRef
    private User userId;

    @DBRef
    private ParkingListing parkingId;

    private VehicleType vehicleType;

    private Date startTime;
    private Date endTime;

    private double totalHours;
    private double totalAmount;

    private BookingStatus status = BookingStatus.BOOKED;

    private String qrCodeDataUrl;

    private Date createdAt = new Date();
    private Date updatedAt = new Date();

    public Booking() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getBookingId() { return bookingId; }
    public void setBookingId(String bookingId) { this.bookingId = bookingId; }

    public User getUserId() { return userId; }
    public void setUserId(User userId) { this.userId = userId; }

    public ParkingListing getParkingId() { return parkingId; }
    public void setParkingId(ParkingListing parkingId) { this.parkingId = parkingId; }

    public VehicleType getVehicleType() { return vehicleType; }
    public void setVehicleType(VehicleType vehicleType) { this.vehicleType = vehicleType; }

    public Date getStartTime() { return startTime; }
    public void setStartTime(Date startTime) { this.startTime = startTime; }

    public Date getEndTime() { return endTime; }
    public void setEndTime(Date endTime) { this.endTime = endTime; }

    public double getTotalHours() { return totalHours; }
    public void setTotalHours(double totalHours) { this.totalHours = totalHours; }

    public double getTotalAmount() { return totalAmount; }
    public void setTotalAmount(double totalAmount) { this.totalAmount = totalAmount; }

    public BookingStatus getStatus() { return status; }
    public void setStatus(BookingStatus status) { this.status = status; }

    public String getQrCodeDataUrl() { return qrCodeDataUrl; }
    public void setQrCodeDataUrl(String qrCodeDataUrl) { this.qrCodeDataUrl = qrCodeDataUrl; }
    
    public Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(Date createdAt) { this.createdAt = createdAt; }
    
    public Date getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Date updatedAt) { this.updatedAt = updatedAt; }
}
